src/gui/optionsdialog.cpp

```diff
@@ -1593,6 +1593,8 @@ int OptionsDialog::getActionOnDblClOnTorrentDl() const
 {
     if (m_ui->actionTorrentDlOnDblClBox->currentIndex() < 1)
         return 0;
```

Comment:
Somewhat related, IMO the above check is out of place (and redundant for this function). All of the boundary checking should occur at: https://github.com/qbittorrent/qBittorrent/blob/be3453684d27e3e308748b021843894c0584e43e/src/gui/optionsdialog.cpp#L1042


This is the same for: https://github.com/qbittorrent/qBittorrent/blob/be3453684d27e3e308748b021843894c0584e43e/src/gui/optionsdialog.cpp#L1604
so if you're interested in cleaning up, those 2 functions should be updated together.